pkg/apply/task/prune_task.go

```diff
@@ -13,9 +14,10 @@ import (
 // by using the PruneOptions. The provided Objects is the
 // set of resources that have just been applied.
 type PruneTask struct {
-	PruneOptions *prune.PruneOptions
-	Objects      []*resource.Info
-	DryRun       bool
+	PruneOptions      *prune.PruneOptions
+	Objects           []*resource.Info
+	DryRun            bool
+	PropagationPolicy metav1.DeletionPropagation
```

Comment:
Should this be separate from the `PruneOptions` or part of it?

Reply:
So we are passing the PropagationPolicy into the `Prune` function as part of the options parameter object. I prefer this solution over setting values directly on the `PruneOptions`. After we moved `DryRun` from being a field on `PruneOptions` to being passed into the `Prune` function, the Applier only interacts with the `PruneOptions` through functions. This means we can use an interface to help with testing.